transform-plugins/src/main/java/co/cask/hydrator/plugin/ValueMapper.java

```diff
@@ -0,0 +1,251 @@
+            "Format is <source-field>:<lookup-table-name>:<target-field>" +
+            "[,<source-field>:<lookup-table-name>:<target-field>]*" +
+            "Source and target field can only of type string." +
+            "For eg: lang_code:language_code_lookup:lang_desc,country_code:country_lookup:country_name")
+    private final String mapping;
+
+    @Name("defaults")
+    @Description("Specify the defaults for source fields if the lookup does not exist or inputs are NULL or EMPTY. " +
+            "Format is <source-field>:<default-value>[,<source-field>:<default-value>]*" +
+            "lang_code:English,country_code:Britain")
+    private final String defaults;
+
+    public Config(String mapping, String defaults) {
+      this.mapping = mapping;
+      this.defaults = defaults;
+    }
+
+    /**
+     * This method is meant to parse input configuration.
+     * It is required to use in configurePiperline as well as transform method.
+     * Hence this is implemented as a part of Config class to set configuration once and make it available for
+     * subsequent methods.
+     */
+    private void parseConfiguration() {
+      Map<String, String> defaultsMapping = new HashMap<>();
+      if (!defaults.isEmpty()) {
+        String[] defaultsList = this.defaults.split(",");
+        for (String defaultValue : defaultsList) {
+          String[] defaultsArray = defaultValue.split(":");
+          defaultsMapping.put(defaultsArray[0], defaultsArray[1]);
+        }
+      }
+      String[] mappingArray = this.mapping.split(",");
+      for (String mapping : mappingArray) {
+        String[] mappingValueArray = mapping.split(":");
+        ValueMapping valueMapping = new ValueMapping();
+        valueMapping.setTargetField(mappingValueArray[2]);
+        valueMapping.setLookupTableName(mappingValueArray[1]);
+
+        if (defaultsMapping.containsKey(mappingValueArray[0])) {
+          valueMapping.setDefaultValue(defaultsMapping.get(mappingValueArray[0]));
+        }
+        mappingValues.put(mappingValueArray[0], valueMapping);
+      }
+    }
+  }
+
+  /**
+   * Creates output schema with the help of input schema and mapping
+   */
+  private Schema getOutputSchema(Schema inputSchema) throws IllegalArgumentException {
+    Schema outputSchema = schemaCache.get(inputSchema);
+    if (outputSchema != null) {
+      return outputSchema;
+    }
+
+    List<Schema.Field> outputFields = new ArrayList<>();
+    for (Schema.Field inputField : inputSchema.getFields()) {
+      if (mappingValues.containsKey(inputField.getName())) {
+        if (inputField.getSchema().getType() != Schema.Type.STRING && !inputField.getSchema().isNullable()) {
+          throw new IllegalArgumentException("Input field " + inputField.getName() + " type should be String");
+        } else {
+          outputFields.add(Schema.Field.of(mappingValues.get(inputField.getName()).getTargetField(),
+                                           inputField.getSchema()));
```

Comment:
the output field schema needs to be a little more complicated.  If there is no default value for the field, the schema for this field should be a nullable string.  If there is a default value for the field, it should just be a string.